src/bin/pgcopydb/ld_apply.c

```diff
@@ -938,6 +952,41 @@ stream_apply_sql(StreamApplyContext *context,
 		}
 
 		case STREAM_ACTION_INSERT:
+		{
+			if (!context->reachedStartPos)
+			{
+				return true;
+			}
+
+			if (context->currentCopyHash == metadata->hash)
+			{
+				return true;
+			}
+			else if (context->currentCopyHash != 0)
+			{
+				/*
+				 * INSERT for a different table, end ongoing COPY and start
+				 * new one.
+				 */
+				pg_copy_end(pgsql);
+				context->currentCopyHash = 0;
+			}
```

Comment:
This takes care of ending the current copy if the new insert targets a different table.